RecoTracker/TkSeedGenerator/src/SeedingOTEDProducer.cc

```diff
@@ -0,0 +1,454 @@
+            if
+              UNLIKELY(!updatedTSOSL1_final.isValid()) continue;
+            std::pair<bool, TrajectoryStateOnSurface> updatedTSOSL2_final =
+                propagateAndUpdate(updatedTSOSL1_final, *buildingPropagator, *hitL2);
+            std::pair<bool, TrajectoryStateOnSurface> updatedTSOSL3_final =
+                propagateAndUpdate(updatedTSOSL2_final.second, *buildingPropagator, hitL3);
+            std::cout << "\t    updatedTSOS final on L3   : " << updatedTSOSL3_final.second << std::endl;
+            TrajectorySeed ts =
+                createSeed(updatedTSOSL3_final.second, container, hitL3.geographicalId(), *buildingPropagator);
+            result.push_back(ts);
+          }
+        }
+
+        std::cout << "-----" << std::endl;
+      }
+    }
+  }
+  std::cout << "-----------------------------" << std::endl;
+  std::cout << "------- seeds found: " << result.size() << " ------" << std::endl;
+  std::cout << "-----------------------------" << std::endl;
+
+  return result;
+}
+
+unsigned int SeedingOTEDProducer::checkLayer(unsigned int iidd) {
+  StripSubdetector strip = StripSubdetector(iidd);
+  unsigned int subid = strip.subdetId();
+  if (subid == StripSubdetector::TIB || subid == StripSubdetector::TOB) {
+    return tkTopo->layer(iidd);
+  }
+  return 0;
+}
+
+std::vector<VectorHit> SeedingOTEDProducer::collectVHsOnLayer(edm::Handle<VectorHitCollectionNew> VHs,
+                                                              unsigned int layerNumber) {
+  const VectorHitCollectionNew& input = *VHs;
+  std::vector<VectorHit> VHsOnLayer;
+  if (!input.empty()) {
+    //std::cout << "initial VH collection size = " << input.size() << std::endl;
+    for (auto DSViter : input) {
+      if (checkLayer(DSViter.id()) == layerNumber) {
+        for (auto vh : DSViter) {
+          VHsOnLayer.push_back(vh);
+        }
+      }
+    }
+  }
+
+  //std::cout << "VH in layer " << layerNumber << " collection size = " << VHsOnLayer.size() << std::endl;
+
+  return VHsOnLayer;
+}
+
+void SeedingOTEDProducer::printVHsOnLayer(edm::Handle<VectorHitCollectionNew> VHs, unsigned int layerNumber) {
+  const VectorHitCollectionNew& input = *VHs;
+  if (!input.empty()) {
+    for (auto DSViter : input) {
+      for (auto vh : DSViter) {
+        if (checkLayer(DSViter.id()) == layerNumber)
+          std::cout << " VH in layer " << layerNumber << " >> " << vh << std::endl;
+      }
+    }
+  } else {
+    std::cout << " No VHs in layer " << layerNumber << "." << std::endl;
+  }
+}
+
+const TrajectoryStateOnSurface SeedingOTEDProducer::buildInitialTSOS(VectorHit& vHit) {
+  //FIXME::charge is fine 1 every two times!!
+  int charge = 1;
+  //float pT = vHit.transverseMomentum(magField);
+  float p = vHit.momentum(magField);
+  float x = vHit.localPosition().x();
+  float y = vHit.localPosition().y();
+  float dx = vHit.localDirection().x();
+  float dy = vHit.localDirection().y();
```

Comment:
this is necer used
please double-check the code for a possible bug
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-0fa1e0/2741/llvm-analysis/report-a02367.html#EndPath

Reply:
I don't understand this error, dy is used in line 348:   
`LocalTrajectoryParameters ltpar2(charge / p, dx, dy, x, y, signPz);`
and is handled exactly the same way as dx for which no error is shown.

Reply:
`dy` is overwritten here 
https://github.com/cms-sw/cmssw/blob/d67dc7f9bb7dcecc93294174955acfb2c03caca4/RecoTracker/TkSeedGenerator/src/SeedingOTEDProducer.cc#L346
(i.e. the value set on the line 331 is never read)

Reply:
Ah, now i get the issue. Thanks! I have reordered the code to avoid the issue.